src/wfpt.pyx

```diff
@@ -70,6 +75,166 @@ def wiener_like(np.ndarray[double, ndim=1] x, double v, double sv, double a, dou
     return sum_logp
 
 
+def wiener_like_rlddm(np.ndarray[double, ndim=1] x,
+                      np.ndarray[long, ndim=1] response,
+                      np.ndarray[double, ndim=1] feedback,
+                      np.ndarray[long, ndim=1] split_by,
+                      double q, double alpha, double dual_alpha, double v, 
+                      double sv, double a, double z, double sz, double t,
+                      double st, double err, int n_st=10, int n_sz=10, bint use_adaptive=1, double simps_err=1e-8,
+                      double p_outlier=0, double w_outlier=0):
+    cdef Py_ssize_t size = x.shape[0]
+    cdef Py_ssize_t i
+    cdef Py_ssize_t s_size
+    cdef int s
+    cdef double p
+    cdef double sum_logp = 0
+    cdef double wp_outlier = w_outlier * p_outlier
+    cdef double alfa
+    cdef double neg_alpha = (2.718281828459**alpha) / (1 + 2.718281828459**alpha)
+    cdef double pos_alpha = 2.718281828459**(alpha + dual_alpha) / (1 + 2.718281828459**(alpha + dual_alpha))
+    cdef np.ndarray[double, ndim=1] feedbacks = np.empty(0, dtype=np.double)
+    cdef np.ndarray[long, ndim=1] responses = np.empty(0, dtype=np.long)
+    cdef np.ndarray[double, ndim=1] xs = np.empty(0, dtype=np.double)
+    cdef np.ndarray[double, ndim=1] qs = np.empty(0, dtype=np.double)
+    cdef np.ndarray[long, ndim=1] unique = np.unique(split_by)
+
+    if not p_outlier_in_range(p_outlier):
+        return -np.inf
+
+    # unique represent # of conditions
+    for s in unique:
+        # select trials for current condition, identified by the split_by-array
+        qs = np.array([q, q])
```

Comment:
Instead of re-initializing `qs` here, can't you move that to the top into a `cdef`? I.e.:
`cdef np.ndarray[double, ndim=2] qs = np.empty([q, q], dtype=np.double)`

And then you would just remove this line and reuse the same array for every iteration of this loop.

Reply:
At that point, the whole function should compile to c.

Reply:
Now defining qs outside the for-loop:
cdef np.ndarray[double, ndim=1] qs = np.array([q, q])